### PR TITLE
Feat/product feed

### PR DESCRIPTION
**Pull request by zweroboy1**
## Describe your changes

- Implemented product feed.

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://2024internship.atlassian.net/browse/SCRUM-71)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [x] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

image

### CODE REVIEW

File: src/main.test.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { act } from 'react';
+import { I18nextProvider } from 'react-i18next';
+import { Provider } from 'react-redux';
+
+import { ThemeProvider } from '@mui/material/styles';
+
+import { render, screen } from '@testing-library/react';
+import { PersistGate } from 'redux-persist/integration/react';
+import GoogleAuthProvider from 'src/components/GoogleAuthProvider';
```
Comment (lyudad) [inline src/main.test.tsx:11]: empty line above

File: src/pages/OnboardingPage/InfoForm/styles.ts [hunk 2]
```diff
@@ -12,4 +15,35 @@ const VisuallyHiddenInput = styled('input')({
...
+  display: 'flex',
+  alignItems: 'center',
+  position: 'relative',
+  border: `1px solid ${theme.palette.border.primary}`,
+  borderRadius: '4px',
+  padding: '9px 4px',
+});
+
+export const StyledPhoneInput = styled(InputBase)({
+  fontFamily: 'DM Sans, Arial, sans-serif',
```
Comment (lyudad) [inline src/pages/OnboardingPage/InfoForm/styles.ts]: 1.3.2

File: src/pages/OnboardingPage/InfoForm/styles.ts [hunk 3]
```diff
@@ -12,4 +15,35 @@ const VisuallyHiddenInput = styled('input')({
...
+  fontSize: 16,
```
Comment (lyudad) [inline src/pages/OnboardingPage/InfoForm/styles.ts]: 1.3.2

File: src/pages/OnboardingPage/index.tsx [hunk 4]
```diff
@@ -5,9 +5,9 @@ import { Typography } from '@mui/material';
 import { Box } from '@mui/system';
 
 import ArrowLeftIcon from 'src/assets/icons/arrow-left.svg';
-import { onboardingSteps } from 'src/common/constants';
+import { onboardingSteps, userRoles } from 'src/common/constants';
 import Header from 'src/components/Header';
-import OnboardingCreditCardForm from 'src/pages/OnboardingPage/CreditCardForm';
+// import OnboardingCreditCardForm from 'src/pages/OnboardingPage/CreditCardForm';
```
Comment (lyudad) [inline src/pages/OnboardingPage/index.tsx]: 1.1.10

File: src/pages/OnboardingPage/index.tsx [hunk 5]
```diff
@@ -38,19 +39,23 @@ function OnboardingPage() {
...
       title: t('onboarding.yourAddress'),
       component: <OnboardingShippingForm />,
-    },
+    } /*
     {
       stepId: onboardingSteps.CARD,
       title: t('onboarding.creditCard'),
       component: <OnboardingCreditCardForm />,
-    },
+    },*/,
```
Comment (lyudad) [inline src/pages/OnboardingPage/index.tsx]: 1.1.10

File: src/components/Header/index.tsx
Comment (lyudad) [inline src/components/Header/index.tsx:1]: need unit test or snapshot its reusable component

File: src/components/ProductCard/index.tsx
Comment (lyudad) [inline src/components/ProductCard/index.tsx:1]: need unit test or snapshot its reusable component

Comment (zweroboy1) [reply src/components/ProductCard/index.tsx:1]: added

File: src/components/Header/index.tsx
Comment (zweroboy1) [reply src/components/Header/index.tsx:1]: added

File: src/pages/OnboardingPage/index.tsx (hunk 5 above)
Comment (zweroboy1) [reply src/pages/OnboardingPage/index.tsx]: fixed

File: src/pages/OnboardingPage/index.tsx (hunk 4 above)
Comment (zweroboy1) [reply src/pages/OnboardingPage/index.tsx]: fixed

File: src/pages/OnboardingPage/InfoForm/styles.ts (hunk 3 above)
Comment (zweroboy1) [reply src/pages/OnboardingPage/InfoForm/styles.ts]: fixed

Comment (zweroboy1) [reply src/pages/OnboardingPage/InfoForm/styles.ts]: fixed

File: src/pages/OnboardingPage/InfoForm/styles.ts (hunk 2 above)
Comment (zweroboy1) [reply src/pages/OnboardingPage/InfoForm/styles.ts]: fixed

File: src/main.test.tsx (hunk 1 above)
Comment (zweroboy1) [reply src/main.test.tsx:11]: fixed

File: src/pages/OnboardingPage/CreditCardForm/index.tsx
Comment (lyudad) [inline src/pages/OnboardingPage/CreditCardForm/index.tsx:1]: you will not use it. this form should came from stripe

Comment (zweroboy1) [reply src/pages/OnboardingPage/CreditCardForm/index.tsx:1]: yes, we did this component, but later desided not to use it

**Pull request merged**